orttraining/orttraining/python/training/optim/config.py

```diff
@@ -0,0 +1,196 @@
+class _OptimizerConfig(object):
+    r"""Base class for optimizer configuration
+
+    This private class is not an optimizer, but a means to configure existing ones from ORT backend.
+    Once the optimizer is configured, no user intervention is needed to update weights or zero gradients during training.
+    The 'parameter group' was inspired by `Pytorch <https://pytorch.org/docs/stable/optim.html#per-parameter-options>`_.
+
+    Args:
+        name (str): optimizer names.
+            One of 'SGDOptimizer', 'AdamOptimizer' and 'LambOptimizer'
+        hyper_parameters (dict): optimizer hyper-parameters applied to all model parameters.
+                                 Every optimizer must have a 'lr' entry on this dictionary.
+        param_groups (list of dict, default is []): list of parameters groups.
+            Each dict must contain a 'params' key with a list of model parameters that will
+            be optimized with the group's custom hyper-parameters values.
```

Comment:
Do we want to include an example here which shows how to freeze some parameters using `param_groups`.

Reply:
That would lead the user to specify 'lr' inside a param group, which is not yet supported. Unless we create a hard coded case which allows 'lr' to be specified if and only if it is zero. Than we would manually feed the weights inside the param group to the `ORTTrainer._internal_use.frozen_weight` attribute

Reply:
Are we supporting freeze parameters through param_groups? or are we keeping the `ORTTrainerOptions._internal_use.frozen_weight` for that?

Reply:
frozen_weight will be part of public API and we won't have explicit freezen weight on param_groups